core/app/models/spree/product.rb

```diff
@@ -36,6 +36,14 @@ class Product < Spree::Base
     has_many :product_promotion_rules, class_name: 'Spree::ProductPromotionRule'
     has_many :promotion_rules, through: :product_promotion_rules, class_name: 'Spree::PromotionRule'
 
+    has_many :promotions, through: :promotion_rules, class_name: 'Spree::Promotion'
+
+    has_many :possible_promotions,
+      -> { advertised.active },
```

Comment:
@Bharat311 Try aligning method arguments by same indentation as first like `->  { advertised.active }` should align with :possible_promotions and respectively other arguments too. Hope this works.